frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-map-manager.ts

```diff
@@ -506,6 +515,54 @@ export class OlMapManager {
         });
     }
 
+    /**
+     * Sets the map's view to see all viewports.
+     */
+    public tryToFitViewToViewports() {
+        if (
+            getSelectRestrictedViewport(this.apiService.ownClientId)(
+                getStateSnapshot(this.store)
```

Comment:
Is this generated multiple times or only once (if multiple times, maybe save it in between)?

Reply:
Caching doesn't make any sense for this (we need the current snapshot value and this operation is only a few lookups -> very cheap).

Reply:
On second thoughts I could make `const currentState = getStateSnapshot(this.store);`.  Because it is used again a few lines below.

I couldn't see this from your comment because you selected different lines...